packages/backend/src/queryBuilder.test.ts

```diff
@@ -133,6 +149,16 @@ describe('Query builder', () => {
         ).toStrictEqual(METRIC_QUERY_WITH_METRIC_FILTER_SQL);
     });
 
+    test('Should build query with metric filter (where filter is disabled) and metric references a dimension from a joined table', () => {
+        expect(
+            buildQuery({
+                explore: EXPLORE,
+                compiledMetricQuery: METRIC_QUERY_WITH_METRIC_DISABLED_FILTER,
+                warehouseClient: warehouseClientMock,
+            }).query,
+        ).toStrictEqual(METRIC_QUERY_WITH_METRIC_FILTER_DISABLED_SQL);
+    });
+
```

Comment:
@ZeRego could this match the scenario you were describing with metric filters + joined tables?